prover/crates/lib/prover_dal/src/fri_witness_generator_dal/node.rs

```diff
@@ -103,22 +105,30 @@ impl FriNodeWitnessGeneratorDal<'_, '_> {
             _ => AggregationRound::NodeAggregation,
         };
 
-        let block_number = L1BatchNumber(row.l1_batch_number as u32);
+        let batch_number =
+            ChainAwareL1BatchNumber::from_raw(row.chain_id as u64, row.l1_batch_number as u32);
+
         let prover_job_ids = self
             .storage
             .fri_prover_jobs_dal()
-            .prover_job_ids_for(block_number, row.circuit_id as u8, round, depth)
+            .prover_job_ids_for(batch_number, row.circuit_id as u8, round, depth)
             .await;
         Some(NodeAggregationJobMetadata {
             id: row.id as u32,
-            block_number,
+            chain_id: batch_number.chain_id(),
+            block_number: batch_number.batch_number(),
             circuit_id: row.circuit_id as u8,
             depth,
-            prover_job_ids_for_proofs: prover_job_ids,
+            prover_job_ids_for_proofs: prover_job_ids.into_iter().map(|ids| ids.1).collect(),
         })
     }
 
-    pub async fn mark_node_aggregation_as_successful(&mut self, id: u32, time_taken: Duration) {
+    pub async fn mark_node_aggregation_as_successful(
+        &mut self,
+        id: u32,
+        chain_id: L2ChainId,
```

Comment:
again, why not use ChainAwareBatchNumber?

Reply:
Because in this case, it is not exactly batch number - it is job id

Reply:
what is a job id in this case? I was thinking now we identify jobs as l1 batch number plus chain id? Or is a different u32?

Reply:
Jobs are identified as L1BatchNumber only for:
* basic circuits WG
* recursion tip WG
* scheduler WG
* compressor
In all other cases they have a different identifier(but yes, it is still a u32). This is because there might be multiple jobs for 1 batch for all the other types of actors